### ISSUE TITLE
Getting "The given key was not present in the dictionary." Error from DescriptionOperationFilter

### ISSUE THREAD

**Issue by ShenglinGuo** [mattfrear/Swashbuckle.Examples#28]
I think I found a bug in the DescriptionOperationFilter. It fails when the model class uses JsonProperty attribute to shape the json result object.

I have a simple response class like following 

```c#
public class AccessTokenResponse
    {
        [JsonProperty("access_token")]
        [Description("Access token from resource owner flow of oauth2, JWT type token")]
        public string AccessToken { get; set; }
        [Description("Authentication type : Bearer")]
        [JsonProperty("token_type")]
        public string TokenType { get; set; }
        [JsonProperty("expires_in")]
        [Description("Access token expires in seconds from issue")]
        public long ExpireInSeconds { get; set; }
        [JsonProperty("refresh_token")]
        [Description("Refresh token to retrieve a new access token")]
        public string RefreshToken { get; set; }
    }
```

I will get "The given key was not present in the dictionary. ..." from DescriptionOperationFilter.

The reason is on line 45
```c#
definition.properties[propertyInfo.Name].description = descriptionAttribute.Description;
```
the propertyInfo.Name is something like AccessToken, but the definition property is expecting access_token.

I have created my own version of DescriptionOperationFilter and change to the RecursivelyParseDescriptions to look like following, it resolves my problem, but do not think this is generic enough for everybody, so just like to report this.

```c#
private static void RecursivelyParseDescriptions(SchemaRegistry schemaRegistry, Type propType)
        {
            Schema definition = schemaRegistry.Definitions[propType.Name];
            foreach (PropertyInfo propertyInfo in propType.GetProperties().Where(prop => prop.IsDefined(typeof(DescriptionAttribute), false)))
            {
                DescriptionAttribute descriptionAttribute = (DescriptionAttribute)propertyInfo.GetCustomAttributes(typeof(DescriptionAttribute), false).First();
                if (definition.properties.ContainsKey(propertyInfo.Name))
                {
                    definition.properties[propertyInfo.Name].description = descriptionAttribute.Description;
                }
                else
                {
                    var jsonPropertyAttr = (JsonPropertyAttribute)propertyInfo.GetCustomAttributes(typeof(JsonPropertyAttribute), false).FirstOrDefault();
                    if (jsonPropertyAttr != null && definition.properties.ContainsKey(jsonPropertyAttr.PropertyName))
                    {
                        definition.properties[jsonPropertyAttr.PropertyName].description = descriptionAttribute.Description;
                    }
                }
            }
            foreach (PropertyInfo propertyInfo in propType.GetProperties().Where(prop => prop.PropertyType.Assembly == propType.Assembly))
            {
                if (schemaRegistry.Definitions.ContainsKey(propertyInfo.Name))
                    RecursivelyParseDescriptions(schemaRegistry, propertyInfo.PropertyType);
            }
        }
```

**Comment by mattfrear** [comment]
Thanks Shenglin. If you get a chance, feel free to submit a PR

**Comment by georgeemr** [comment]
Hi, im facing the same problem but i'm not using the AspNetCore version. Can someone pull this to this current project.

**Comment by mattfrear** [comment]
You're welcome to do it @georgeemr . I'll do it when I get a chance, possibly some time in the next week.

**Comment by mattfrear** [comment]
I have ported the referenced PR from Swashbuckle.AspNetCore.Examples over and released a NuGet package 3.6.1.

https://www.nuget.org/packages/Swashbuckle.Examples/3.6.1

Closing this issue. Please reopen if it's not fixed for you @georgeemr
